kivy/uix/switch.py

```diff
@@ -73,19 +143,188 @@ class Switch(Widget):
     '''(internal) Contains the distance between the initial position of the
     touch and the current position to determine if the swipe is from the left
     or right.
-
     :attr:`touch_distance` is a :class:`~kivy.properties.NumericProperty`
     and defaults to 0.
     '''
 
     active_norm_pos = NumericProperty(0)
     '''(internal) Contains the normalized position of the movable element
     inside the switch, in the 0-1 range.
-
     :attr:`active_norm_pos` is a :class:`~kivy.properties.NumericProperty`
     and defaults to 0.
     '''
 
+    switch_background_relative_size_hint_y = NumericProperty(1, allownone=True)
```

Comment:
I think the names can be simplified a bit. E.g. for all the new properties, please remove the `switch_` prefix. The class is a switch so there's no need to name the properties with the class name.

Also, you can remove the `relative` part from these properties. So e.g. this would just be `background_size_hint_y`.

Also, I'm not sure that using a size hint for an image is a good idea. We don't generally do this in Kivy because images are generally designed for a fixed size (e.g. the text would look weird if scaled up). So it may be better to not add these size hints.

Also, I saw that you removed some blank lines in the docs, but I'm not sure why? They are needed there to properly divide the docs.

Reply:
I agree with the names, I will simplify them.

As for the blank lines, I removed them at the beginning of the edition and ended up not realizing that they were necessary when I finished. As I couldn't test it, I didn't see that it would be missed. But I will add them again.

Regarding the size hint, it was a way I found to make the Switch components honor the properties of the Widget class. Because in my view, as all Switch components are images, having a size hint for the Switch widget itself, while its components are not responsive, seems a bit unexpected. What do you think about that?

Reply:
>Because in my view, as all Switch components are images, having a size hint for the Switch widget itself, while its components are not responsive, seems a bit unexpected. What do you think about that?

I do understand that. But I wonder if anyone uses icon images with a size hint. Personally, all my buttons and icons use a fixed size because making it responsive just didn't look right, unless I switch images depending on the size. So while what you're saying is right, if people don't use size hints, it may not be worth adding support for it, even the widget technically supports it. 

A user could always manually set the image size based on the widget size so it would be too hard for them to make it responsive. E.g.: `button_size: self.size` or something. But I'd be interested to hear if there are uses cases that use size hints for icons.

Reply:
Really, using size hint to scale icons can be risky.
I remember using size hint on icons when I started using kivy. I defined size_hint_x, for example, and associated the height to it, obeying the actual proportion of the image. It worked fine, but everything changed when I found out that using density-independent pixels would give me more control.

Well, I think it’s better then to remove those size hints.
Soon I will commit with the changes.